pkg/services/authz/rbac/service_test.go

```diff
@@ -1191,6 +1191,121 @@ func TestService_List(t *testing.T) {
 	})
 }
 
+func TestService_getAnonymousPermissions(t *testing.T) {
+	type testCase struct {
+		name          string
+		permissions   []accesscontrol.Permission
+		action        string
+		actionSets    []string
+		cacheHit      bool
+		expectedPerms map[string]bool
+		expectedError bool
+		anonRole      string
+	}
+
+	testCases := []testCase{
+		{
+			name: "should return permissions from cache if available",
+			permissions: []accesscontrol.Permission{
+				{Action: "dashboards:read", Scope: "dashboards:uid:some_dashboard"},
+			},
+			action:        "dashboards:read",
+			actionSets:    []string{},
+			cacheHit:      true,
+			expectedPerms: map[string]bool{"dashboards:uid:some_dashboard": true},
+			expectedError: false,
+			anonRole:      "Viewer",
+		},
+		{
+			name: "should return permissions from store if not in cache",
+			permissions: []accesscontrol.Permission{
+				{Action: "dashboards:read", Scope: "dashboards:uid:some_dashboard"},
+			},
+			action:        "dashboards:read",
+			actionSets:    []string{},
+			cacheHit:      false,
+			expectedPerms: map[string]bool{"dashboards:uid:some_dashboard": true},
+			expectedError: false,
+			anonRole:      "Viewer",
+		},
+		{
+			name:          "should return error if store fails",
+			permissions:   nil,
+			action:        "dashboards:read",
+			actionSets:    []string{},
+			cacheHit:      false,
+			expectedPerms: nil,
+			expectedError: true,
+			anonRole:      "Viewer",
+		},
+		{
+			name: "should handle wildcard permissions",
+			permissions: []accesscontrol.Permission{
+				{Action: "dashboards:read", Scope: "*", Kind: "*"},
+			},
+			action:        "dashboards:read",
+			actionSets:    []string{},
+			cacheHit:      false,
+			expectedPerms: map[string]bool{"*": true},
+			expectedError: false,
+			anonRole:      "Viewer",
+		},
+		{
+			name: "should use custom anonymous role when specified",
```

Comment:
I feel like we're not really testing that the role gets set correctly. Maybe you could set the role in config, and then check that when authZ service is initialized by calling `ProvideAuthZClient ` the role gets set correctly?

I don't want to block on this though. I think it just shows that we probably need some integration tests for authZ service to properly test the different flows with real stores etc.